src/main/java/randoop/reflection/SignatureParser.java

```diff
@@ -0,0 +1,152 @@
+package randoop.reflection;
+
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parses type signature strings used to identify methods and constructors in input.
+ *
+ * <p>Note: this duplicates should be factored into a separate source set (aka, module) so that it
+ * can also be used in javagents. The patterns are duplicated from {@code ReplacementFileReader}
+ * from the mapcall agent.
+ */
+public class SignatureParser {
+  /** Regex for Java identifiers */
+  private static final String ID_STRING =
+      "\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*";
+
+  /**
+   * Regex to match a sequence of identifiers separated by periods. Corresponds to package names,
+   * fully-qualified classnames, or method names with fully-qualified classname.
+   */
+  private static final String DOT_DELIMITED_IDS =
+      ID_STRING + "(?:" + "\\." + "(?:" + ID_STRING + "|" + "<init>" + "))*";
+
+  /**
+   * Naive regex to match a method signature consisting of a fully-qualified classname, a method
+   * name, and argument types in parentheses. The pattern does not include the details of the
+   * argument types.
+   */
+  private static final Pattern SIGNATURE_PATTERN =
+      Pattern.compile("(" + DOT_DELIMITED_IDS + ")\\(([^)]*)\\)");
+
+  /**
+   * Parses a fully-qualified signature and returns the corresponding {@code
+   * java.lang.reflect.AccessibleObject}.
+   *
+   * <p>A signature is expected to have the form
+   *
+   * <ul>
+   *   <li>{@code package-name.classname.method-name(argument-list)} for a method,
+   *   <li>{@code package-name.classname.&lt;init&gt;(argument-list)} or {@code
+   *       package-name.classname(argument-list)} for a constructor.
+   * </ul>
+   *
+   * <p>where {@code package-name} is a period-separated list of identifiers, and <code>
```

Comment:
Use `{@code ...}` instead of `<code>...</code>`.  It's shorter, is usually easier to read, and suffers fewer formatting artifacts (such as extra spaces before or after) in the resulting HTML.